modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -283,6 +284,12 @@ class PublishDir {
         this.stageInMode = task.config.stageInMode
         this.taskName = task.name
 
+        // resolve Fusion symlinks
+        if( FusionHelper.isFusionEnabled(Global.session as Session) ) {
+            final inputFiles = task.getInputFilesMap()
+            files = files.collect { inputFiles.getOrDefault(it.name, it) } as Set<Path>
+        }
+
```

Comment:
Not sure to understand how this works?

Reply:
Probably was easier to read before I compacted it into three lines 😄 

When Fusion is enabled, if an output file is found to be an input, then we know that it is a Fusion symlink, so we should instead publish the link source, which we can obtain by reading the link file from S3 or from the input files map. Obviously the second way is more efficient.

The input files maps stage name to source path, so we check each output file to see if it's name is in the map, and if it is then replace the output file with the source path.